src/synthesizer.ts

```diff
@@ -212,6 +250,63 @@ export class PulumiSynthesizer extends PulumiSynthesizerBase implements cdk.IReu
         return id;
     }
 
+    /**
+     * This will create a unique ECR repository for each asset. Creating a separate repository for each
+     * asset allows us to have a lifecycle policy on the repository to delete old images.
+     *
+     * @param asset - The cdk asset to create a repo for
+     * @returns Information about the created repo
+     */
+    private getCreateRepo(asset: cdk.DockerImageAssetSource): CreateRepoResponse {
+        if (!asset.assetName) {
+            throw new Error("Docker image assets must include 'assetName' in the asset source definition");
+        }
+        const repoName = `${this.appId}/${asset.assetName}`
+            .toLocaleLowerCase()
+            .replace('.', '-')
+            // it can only start with letters or numbers
+            .replace(/[^a-z0-9]/, '');
+
+        if (!this.stagingRepos[repoName]) {
+            const repo = new aws.ecr.Repository(
+                repoName,
+                {
+                    name: repoName,
+                    // prevents you from pushing an image with a tag that already exists in the repository
+                    // @see https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-tag-mutability.html
+                    imageTagMutability: 'IMMUTABLE',
```

Comment:
Is this also immutable on the CDK side?
Given that the tag should only change if the image changes, I think we should be good with tag immutability but I wonder if this is the expected config for CDK users

Reply:
Yep this is the config on the CDK side as well

https://github.com/aws/aws-cdk/blob/ccab485b87a7090ddf0773508d7b8ee84ff654b0/packages/@aws-cdk/app-staging-synthesizer-alpha/lib/default-staging-stack.ts?plain=1#L453